HW2/HW2_Khyzhniak.js

```diff
@@ -0,0 +1,588 @@
+  
+  var varExample = 'Awesome var!';
+  let letExample = 'Awesome let!';
+  const constExample = 'Awesome const!';
+  
+  console.log(varExample); // WORK`S FINE - 'Awesome var!'
+  console.log(letExample); // WORK`S FINE - 'Awesome let!'
+  console.log(constExample); // WORK`S FINE - 'Awesome const!'
+  /* SUMMARY: variables created with var are undefined from the beginning of the function */
+}
+
+// 1.6 Const with no value
+// IMAGINE, on early Monday morning you came to the office and created a constant with no value because you don`t know yet what the value is..
+function example1_6_0() {
+  const someVariable;
+  // That`s useless, you can`t assign to constant variable.
+  /* SUMMARY: Don`t come to the office on Monday.
+    Just joking, don`t declare constants with no value. */
+}
+
+// 1.7 Shorter syntax
+function example1_7_0() {
+let a = 1;
+let b = 2;
+let c = 3;
+let d = 4;
+// Instead use:
+let a = 1, b = 2, c = 3, d = 4;
+// NOTE, it`s harder to read!
+}
+
+// 1.8 Additional info about variables
+// 1.8.0 Always use camelCase for naming a variable in JavaScript. That`s a good practise in JavaScript.
+function example1_8_0() {
+  var dmytroKhyzhniak;
+  let coolDmytroKhyzhniak;
+  const dmytroKorol;
+}
+
+// 1.8.1 In other languages you can see PascalCase, snake_case, kebab-case.
+function example1_8_1() {
+  var ThisIsPascalCase;
+  let this_is_snake_case;
+}
+
+// 1.8.2 You can use $ and _ for naming variables
+function example1_8_2() {
+  var _ = 'Awesome!';
+  let $ = 'Awesome!';
+}
+
+// 1.8.3 You can`t use special keywords for naming a variables
+function example1_8_3() {
+  let this; // ERROR | Unexpected token 'this'
+  var function; // ERROR | Unexpected token 'function'
+}
+
+// 1.8.4 You can`t start naming a variable with number or other symbols exept those two
+function example1_8_4() {
+  var %variable; // ERROR | Unexpected token '%'
+  let 3variable; // ERROR | Invalid or unexpected token
+}
+
+// 1.8.5 You can use non-latin symbols but that`s a bad practise
+function example1_8_5() {
+  var змінна;
+  let переменная;
+}
+
+// 2. -------- Write a good example of how Hoisting works --------
+
+// IMAGINE WE WANT TO BUILD A HOUSE WITH JAVASCRIPT..
+// YUP, THE FLOORS WILL BE UPSIDE-DOWN, NEVERMIND. JUST IMAGINE!
+// EVERY BLOCK WILL BE BUILD FROM DIFFERENT MATERIALS - VAR, LET, CONST OR EVEN FUNCTIONS!
+
+function example2_0_0() {
+  // BUILDING STRARTED 🕐
+                     let  floor1 = 'Floor #1 | □ □ | LET';
+                   const  floor2 = 'Floor #2 | □ □ | CONST';
+                     var  floor3 = 'Floor #3 | □ □ | VAR';
+  let floor4 = function() { return 'Floor #4 | □ □ | LET FUNCTION EXPESSION' };
+const floor5 = function() { return 'Floor #5 | □ □ | CONST FUNCTION EXPESSION' };
+  var floor6 = function() { return 'Floor #6 | □ □ | VAR FUNCTION EXPESSION' };
+        function floor7() { return 'Floor #7 | □ □ | FUNCTION DECLARATION' };
+  // ✅ BUILDING COMPLETED ✅
```

Comment:
I got the idea, but it's better to use formatting by the first character in the line)